biothings/hub/dataload/uploader.py

```diff
@@ -429,11 +429,16 @@ def register_status(self, status, subkey="upload", **extra):
         # with arguments like 'init' and 'transient'...
         if status.endswith("ing"):
             # record some "in-progress" information
-            upload_info['step'] = self.name  # this is the actual collection name
-            upload_info['temp_collection'] = self.temp_collection_name
-            upload_info['pid'] = os.getpid()
-            upload_info['logfile'] = self.logfile
-            upload_info['started_at'] = datetime.datetime.now().astimezone()
+
+            upload_info["step"] = self.name  # this is the actual collection name
+            upload_info["temp_collection"] = self.temp_collection_name
+            upload_info["pid"] = os.getpid()
+            upload_info["logfile"] = self.logfile
+            upload_info["started_at"] = datetime.datetime.now().astimezone()
+
+            # We should use the last_success from the last upload time as a default value for the current's last_success.
+            last_upload_info = self.src_doc.get(subkey, {}).get("jobs", {}).get(self.name, {})
+            upload_info["last_success"] = last_upload_info.get("last_success")
```

Comment:
same here, what if there is no "last_success" in `last_upload_info`?

Reply:
Same as the `last_success` in `last_download_info`, if there is no information we want, this code will not failed, and the `upload_info["last_success"]` will be set to None.